vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go

```diff
@@ -56,25 +68,39 @@ type prometheusHistoryProvider struct {
 	prometheusClient PrometheusClient
 }
 
+var singletonHistoryProvider *prometheusHistoryProvider
+
 // NewPrometheusHistoryProvider contructs a history provider that gets data from Prometheus.
-func NewPrometheusHistoryProvider(prometheusAddress string) HistoryProvider {
-	return &prometheusHistoryProvider{
-		prometheusClient: NewPrometheusClient(&http.Client{}, prometheusAddress),
-	}
+func NewPrometheusHistoryProvider(config PrometheusHistoryProviderConfig) HistoryProvider {
+	hpConfig = prometheusHistoryProviderConfig{
+		historyLength:          config.HistoryLength,
+		podLabelPrefix:         config.PodLabelPrefix,
+		podLabelsMetricName:    config.PodLabelsMetricName,
+		podNamespaceLabel:      config.PodNamespaceLabel,
+		podNameLabel:           config.PodNameLabel,
+		ctrNamespaceLabel:      config.CtrNamespaceLabel,
+		ctrPodNameLabel:        config.CtrPodNameLabel,
+		ctrNameLabel:           config.CtrNameLabel,
+		cadvisorMetricsJobName: config.CadvisorMetricsJobName,
+	}
+	singletonHistoryProvider = &prometheusHistoryProvider{
+		prometheusClient: NewPrometheusClient(&http.Client{}, config.Address),
+	}
+	return singletonHistoryProvider
 }
 
 func getContainerIDFromLabels(labels map[string]string) (*model.ContainerID, error) {
```

Comment:
If you make config part of prometheusHistoryProvider, you can make these functions methods of *prometheusHistoryProvider. For example here 
`func(p *prometheusHistoryProvider) getContainerIDFromLabels(labels map[string]string) (*model.ContainerID, error) {`

and then use `p.config.sth`

Reply:
done